pep-XXXX.rst

```diff
@@ -0,0 +1,492 @@
+  - The ``__repr__`` methods will differ and display the new builtin syntax
+  - But the ``__args__`` and ``__parameters__`` fields must behave the same
+  - And the indexing operation - which returns a new type object with concrete types substituted for various entries in ``__parameters__``, must also be the same.
+
+We will return to more details of the runtime behavior, which remain open to discussion other than backward compatibility, in the Open Issues section below.
+
+
+Open Issues
+===========
+
+Details of the Runtime API
+--------------------------
+
+The new runtime objects to which this syntax evaluates will remain backward-compatible with the ``typing.Callable`` and ``typing.Concatenate`` types they replace, other than details like ``__repr__`` where some behavior change makes sense.
+
+But we also believe that we should have a new runtime API with more structured data access, since:
+- Callable types have a more complicated shape than other generics, especially given the behavior when using ``...`` and ``typing.Concatenate``
+- In the future we might want to add more features, such as support for named and optional arguments, that would be even more difficult to describe well using only ``__args__`` and ``__parameters___``.
+
+Our tentative plan is to define enough new builtins for the runtime data to mirror the shape of the AST, but other options are also possible. See [#runtime-behavior-specification]_ for a detailed description of the current plan and a place to discuss other ideas.
+
+Once the runtime behavior is fully defined we will add a complete evaluation model and description of behavior to this PEP.
+
+Optimizing ``SyntaxError`` messages
+-----------------------------------
+
+The current reference implementation has a fully-functional parser and all edge cases presented here have been tested.
+
+But there are some known cases where the errors are not as informative as we would like. For example, because ``(int, ...) -> bool`` is illegal but ``(int, ...)`` is a valid tuple, we currently produce a syntax error flagging the ``->`` as the problem even though the real cause of the error is using ``...`` as an argument type.
+
+This is not part of the specification per se but is an important detail to add  ress in our implementation. The solution will likely involve adding ``invalid_.*`` rules to ``python.gram`` and customizing error messages.
+
+Resources
+=========
+
+Background and History
+----------------------
+
+PEP 484 [#pep-484-function-type-hints]_ specifies a very similar syntax for function type hint *comments* for use in code that needs to work on Python 2.7, for example::
+
+    def f(x, y):
+        # type: (int, str) -> bool
+        ...
+
+At that time we used indexing operations to specify generic types like ``typing.Callable`` because we decided not to add syntax for types, but we have since begun to do so, e.g. with PEP 604
```

Comment:
```suggestion
At that time we used indexing operations to specify generic types like ``typing.Callable`` because we decided not to add syntax for types. However, we have since begun to do so, e.g. with PEP 604.
```